### PR TITLE
ci: automated NuGet publish and GitHub release integration/setup

### LINKED ISSUE #54: Add automated publish of the NuGet packages
# Goal

Allow for automated publish of NuGet packages, so we can continuously publish packages

# Acceptance Criteria

*  Packages are pushed to Nuget via GitHub workflow
*  Nuget shows now errors with the packages

### LINKED ISSUE #70: Add automated GitHub Releases

### PR DESCRIPTION
**Pull request by PeterMarinov**
# Automated NuGet publish and GitHub release integration/setup

## Description
* Adds automated release to NuGet via GitHub Actions
* Adds ability to propagate GitHub release tags to the NuGet and MsBuild versions
* Automated release to NuGet based of pushing GitHub release tags. The following format is supported:
           * `v{major}.{minor}.{patches}` - official releases
           * `v{major}.{minor}.{patches}-alpha` - Alpha prerelease
           * `v{major}.{minor}.{patches}-beta` - Beta prerelease
           * `v{major}.{minor}.{patches}-rc` - Release candidate prerelease

## Related Issues
closes #54 
closes #70 

## Type of Change
Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update - note, docs will be added to external KB

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots
If applicable, add screenshots to help explain your changes.

image

image

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-alpha'
+      - 'v[0-9]+.[0-9]+.[0-9]+-beta'
+      - 'v[0-9]+.[0-9]+.[0-9]+-rc'
+
+jobs:
+  release-version:
+    env:
+      PACKAGE_ID: "KungFunc.CLI.Demo"
```
Comment (dspetrov) [inline .github/workflows/release.yml:23]: Is it intentional that the name is `KungFunc`?

Comment (PeterMarinov) [reply .github/workflows/release.yml:23]: Nope...this has to be dynamic or removed...Let me check

Comment (PeterMarinov) [reply .github/workflows/release.yml]: @dspetrov  - good catch. This is left-over from the Demo repo testing. 

Also forgot to finish the renaming of the Endatix.SqlServer to Endatix.Persistence.SqlServer as agreed previously. Now it's done

Another issue I have found was that the Endatix.Api.Host was not added to the solution, which I have fixed and it's OK now

**Review by dspetrov — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (dspetrov) [reply .github/workflows/release.yml]: Great! All seems fine now.

**Review by dspetrov — APPROVED** [review]

**Pull request merged**